transport/src/main/java/io/camunda/zeebe/transport/stream/impl/messages/AddStreamResponse.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.transport.stream.impl.messages;
+
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+
+public final class AddStreamResponse implements StreamResponse {
+  private final MessageHeaderEncoder headerEncoder = new MessageHeaderEncoder();
+  private final MessageHeaderDecoder headerDecoder = new MessageHeaderDecoder();
+
+  private final AddStreamResponseEncoder messageEncoder = new AddStreamResponseEncoder();
+  private final AddStreamResponseDecoder messageDecoder = new AddStreamResponseDecoder();
+
+  @Override
+  public void wrap(final DirectBuffer buffer, final int offset, final int length) {
+    messageDecoder.wrapAndApplyHeader(buffer, offset, headerDecoder);
+  }
+
+  @Override
+  public int getLength() {
+    return headerEncoder.encodedLength() + messageEncoder.sbeBlockLength();
```

Comment:
❓ no encoded length? I'm always confused by this, and don't understand why we need to write these classes our own. Feels always error prone 😅 Didn't we had also a while ago a super class for such things?

Reply:
Ah ok it is because the messages are all empty. Is this safe to change later? Or should we already have some like error code, metadata, message which are optional? 🤔

Reply:
Hm, that's a different way than I expected. You mean remove the `ErrorResponse` message, and instead have all responses possibly carry an error as well? Would be nice if SBE supported `oneof` kind of structure, but as it is, I think two messages is cleaner.

As for evolving, SBE supports adding optional fields or version-tagged fields. So if we had a fixed-length field with `since=2`, for example, it will know not to look for it/skip it/omit it from the length when the local protocol version is 1.

What it doesn't support is optional data (i.e. variable length fields). Once you add data, it's either always there, or you have to explicitly skip it. But since data is always serialized at the end of messages, it's always backwards compatible (i.e. an older node would simply stop reading before then). Does that make sense?